### PR TITLE
Feature/api flight deals

### PR DESCRIPTION
**Pull request by bardiamazaheri**

### CODE REVIEW

File: expo/screens/GoScreen.js [hunk 1]
```diff
@@ -234,7 +239,17 @@ function GoScreen({ navigation, initializeForm, updateIsFrom, updateIsTo }) {
...
+  } else if(currentSection === 'Flight Deals Details') {
+    content = <FlightDealsDetailsScreen navigation={navigation} page={page} />
+  } else if(currentSection === 'Flight Deals Review') {
+    content = <FlightDealsReviewScreen navigation={navigation}/>
+  } else if(currentSection === 'Sightseeing Details') {
+    content = <SightseeingDetailsScreen navigation={navigation} />
+  } else if(currentSection === 'Sightseeing Intro') {
+    content = <SightseeingIntroScreen navigation={navigation} />
+  } else if(currentSection === 'Sightseeing Review') {
+    content = <SightseeingReviewScreen navigation={navigation} />
```
Comment (jet2go) [inline expo/screens/GoScreen.js]: @bardiamazaheri What about Empty Legs pages?

File: expo/screens/NearbyAirportsScreen.js [hunk 2]
```diff
@@ -26,22 +27,6 @@ const styles = StyleSheet.create({
...
-      timeout = null;
-      if (!immediate) func.apply(context, args);
-    };
-    const callNow = immediate && !timeout;
-    clearTimeout(timeout);
-    timeout = setTimeout(later, wait);
-    if (callNow) func.apply(context, args);
-  };
-}
-
```
Comment (jet2go) [inline expo/screens/NearbyAirportsScreen.js]: Revert this part as it's needed for the map rendering.

File: expo/screens/SightseeingDetailsScreen.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                          textAlignVertical: 'center',
+                        }}
+                      >
+                        Continue
+                      </Text>
+                    </TouchableOpacity>
+                  </View>
+                )) ||
+                  null}
+
```
Comment (jet2go) [inline expo/screens/SightseeingDetailsScreen.js]: nit: remove empty line

File: expo/components/FlightListItem/EmptyLegListItem.js [hunk 4]
```diff
@@ -3,42 +3,42 @@ import {connect} from 'react-redux';
...
             </View>
-        </View>
+        </TouchableOpacity>
     );
 }
 
-export default connect(state => ({
-
-}))(EmptyLegListItem);
+export default connect(state => ({}))(EmptyLegListItem);
```
Comment (xjlim) [inline expo/components/FlightListItem/EmptyLegListItem.js]: if redux is not needed, can remove the connect wrapper

File: expo/screens/EmptyLegsScreen.js [hunk 5]
```diff
@@ -1,70 +1,101 @@
...
 import { EmptyLegListItem } from '../components/FlightListItem'
+import {ApiUrl} from "../constants/index";
+import { SearchBar } from 'react-native-elements';
+import {updateAircraft, updateDate, updateFrom, updatePassengersCount, updateTo} from "../actions/updateFlights";
+import {updateDealInfo} from "../actions/updateDeals";
+import debounce from "../components/Debounce";
 
-function EmptyLegsScreen(props) {
+async function getDealsAsync(token, searchValue, setDeals) {
+  const resp = await fetch(`${ApiUrl}/api/deal?dealType=Empty Leg&searchTerm=${searchValue}`, {
```
Comment (xjlim) [inline expo/screens/EmptyLegsScreen.js]: can dealtype not have spaces? url typically do not contain spaces (they are encoded)

**Pull request merged**